t/unit/transport/test_librabbitmq.py

```diff
@@ -125,6 +125,17 @@ def test_collect__with_conn(self):
             self.T._collect(conn)
             close.assert_called_with(conn.fileno())
 
+    def test_collect__with_fileno_raising_value_error(self):
+        conn = Mock(name='connection')
+        conn.channels = {1: Mock(name='chan1'), 2: Mock(name='chan2')}
+        with patch('os.close') as close:
+            self.T.client = self.client
+            conn.fileno.side_effect = ValueError("Socket not connected")
+            self.T._collect(conn)
+            close.assert_not_called()
```

Comment:
Can you please also add an assertion that `conn.fileno` has actually been called once?

I believe that your patch and test case is 100% correct, but perhaps there is an issue with the test that also causes Code Coverage to be not reported as expected. What do you think?

Reply:
Thanks @georgepsarakis I gave asserting fileno was called a try and that didn't seem to work, but was definitely worth a shot.

Reply:
@bshelton229 thanks for trying this, I believe this is ready for merge.